odl/solvers/nonsmooth/douglas_rachford.py

```diff
@@ -224,3 +231,100 @@ def douglas_rachford_pd(x, f, g, L, tau, sigma, niter,
     # The final result is actually in p1 according to the algorithm, so we need
     # to assign here.
     x.assign(p1)
+
+
+def _operator_norms(L):
+    """Get operator norms if needed.
+
+    Parameters
+    ----------
+    L : sequence of `Operator` or float
+        The operators or the norms of the operators that are used in the
+        `douglas_rachford_pd` method. For `Operator` entries, the norm
+        is computed with `power_method_opnorm`.
+    """
+    L_norms = []
+    for Li in L:
+        if np.isscalar(Li):
+            L_norms.append(float(Li))
+        elif isinstance(Li, Operator):
+            L_norms.append(Li.norm(estimate=True))
+        else:
+            raise TypeError('invalid entry {!r} in `L`'.format(Li))
+    return L_norms
+
+
+def douglas_rachford_pd_stepsize(L, tau=None, sigma=None):
+    r"""Default step sizes for `douglas_rachford_pd`.
+
+    Parameters
+    ----------
+    L : sequence of `Operator` or float
+        The operators or the norms of the operators that are used in the
+        `douglas_rachford_pd` method. For `Operator` entries, the norm
+        is computed with `Operator.norm(estimate=True)`.
+    tau : positive float, optional
+        Use this value for ``tau`` instead of computing it from the
+        operator norms, see Notes.
+    sigma : tuple of float, optional
+        The ``sigma`` step size parameters for the dual update.
+
+    Returns
+    -------
+    tau : float
+        The ``tau`` step size parameter for the primal update.
+    sigma : tuple of float
+        The ``sigma`` step size parameters for the dual update.
+
+    Notes
+    -----
+    To guarantee convergence, the parameters :math:`\tau`, :math:`\sigma_i`
+    and :math:`L_i` need to satisfy
+
+    .. math::
+       \tau \sum_{i=1}^n \sigma_i \|L_i\|^2 < 4.
+
+    This function has 4 options, :math:`\tau`/:math:`\sigma` given or not
```

Comment:
Maybe use an itemize list to improve readability? Like
```
- Neither :math:`\tau` nor :math:`\sigma` given:
   
   .. math::
       ...
```

Reply:
Great suggestion, fixing that.